### DISCUSSION TITLE
AI Powered Microsoft Terminal with GPT 5/4?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/terminal#15244] [Ideas]
Like warp.dev, It's a GPT 3 powered AI Terminal.
What's we need is microsoft power the terminal with the GPT 4 / 5 generation AI.

**Comment by nguyen-dows** [comment]
Hi @cnmade! This sounds like an interesting idea. What would you use an AI in Windows Terminal for?

**Comment by ghost** [comment]
Short say: every things.

AI can answer your question, can auto complete your promt input, can fix your typo mistake.

can help you write your shell script to do some jobs.

Like warp teminal:

Explain output from the terminal
Suggest how to fix an error (Warp AI will automatically detect errors in output)
Write a script for you (e.g. write an automated script to ssh into a cloud instance)
Walk you through a setup (e.g. install latest version of postgres)
Show examples

**Comment by cornradio** [comment]
the best thing about warp is not : asking for commands

is that warp will know what you wanna type after you type half of them !
